src/test/ui/nll/type-check-pointer-coercions.stderr

```diff
@@ -39,6 +45,12 @@ LL | fn unique_to_mut<'a, 'b>(x: &mut &'a i32) -> *mut &'b i32 {
 LL |     // Two errors because *mut is invariant
 LL |     x
    |     ^ returning this value requires that `'a` must outlive `'b`
+   |
+   = help: consider adding the following bound: `'a: 'b`
+
+help: the following changes may resolve your lifetime errors
+   |
+   = help: `'b` and `'a` must be the same; replace one with the other
```

Comment:
I think this case would be better served with not emitting the two previous errors and emit a single error stating the information from this `help`. That being said I'll take a look at the code to see how hard it would be to do that, but tentatively r=me as is.

Reply:
Hmm... This might require a bit of work. Currently, we don't discover that they are the same until after we have buffered the two previous errors.